src/HTTP/Exchange/Request.cs

```diff
@@ -7,14 +8,25 @@ public struct Request
         public readonly string Method;
         public readonly string Path;
         public readonly string Protocol;
-        public readonly Headers _headers;
+        private readonly Headers _headers;
+        public readonly string Body;
 
         public Request(string method, string path, string protocol)
```

Comment:
Could you chain the constructors to remove the duplicated logic?

Reply:
And possibly reorder them so that the constructor it calls appears next in the code listing?